packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -464,4 +468,31 @@ export default class {
       findKey(this._xapis, candidate => candidate === sourceXapi)
     )::ignoreErrors()
   }
+
+  async detachHostFromPool(hostId) {
+    const xapi = this.getXapi(hostId)
+    const { address } = xapi.getObject(hostId)
+
+    await xapi.ejectHostFromPool(hostId)
+
+    this._getXenServer(
+      findKey(this._xapis, candidate => candidate === xapi)
+    ).then(
+      ({ properties }) =>
+        this.registerXenServer({
+          ...properties,
+          host: address,
+        }).then(({ id }) => {
+          const loop = () =>
+            pDelay(6e4).then(() =>
+              this.connectXenServer(id).catch(
+                ({ message }) =>
+                  message === CONNECTION_TO_CONNECTED_POOL_ERROR && loop()
+              )
+            )
+          return loop()
+        }),
+      () => {}
```

Comment:
Use `ignoreErrors` instead because you might swallow some programmer errors here.